docs/testing/e2e/mobile-e2e-mocking.md

```diff
@@ -0,0 +1,176 @@
+│   ├── api-mocking/
+│       ├── api-mocking/
+│       │   ├── mock-responses/
+│       │   │   ├── gas-api-responses.json
+```
+This structure promotes clear organisation and makes managing tests simpler.
+
+## Mock Server Implementation
+
+### Overview
+
+We use Mockttp to simulate API responses, providing flexible testing across different HTTP methods. This allows us to test app behaviour even when external dependencies are unavailable or unreliable.
+
+### Key Features
+
+- Supports multiple HTTP methods (GET, POST, etc.)
+- Configurable requests and responses
+- Logs responses to simplify debugging
+
+### Naming Mock Test Files
+
+Mock test files are named with `.mock.spec.js` to keep things organised. For example, a test for the suggested gas API would be named: `suggestedGasApi.mock.spec.js`.
+
+### Setting Up the Mock Server
+
+The `startMockServer` function in `e2e/api-mocking/mock-server.js` starts the mock server. It takes events organised by HTTP methods, specifying the endpoint, response data, and request body (for POST requests).
+
+```javascript
+import { mockEvents } from '../api-mocking/mock-config/mock-events';
+
+mockServer = await startMockServer({
+  GET: [
+    mockEvents.GET.suggestedGasApiErrorResponse,
+  ],
+  POST: [
+    mockEvents.POST.suggestedGasApiPostResponse,
+  ],
+});
+```
+
+### Defining Mock Events
+
+`mockEvents.js` defines mock events, including:
+
+- `urlEndpoint`: The API endpoint being mocked
+- `response`: The mock response the server will return
+- `requestBody`: Expected request body (for POST requests)
+
+```javascript
+export const mockEvents = {
+  GET: {
+    suggestedGasApiErrorResponse: {
+      urlEndpoint: 'https://gas.api.cx.metamask.io/networks/1/suggestedGasFees',
+      response: { status: 500, message: 'Internal Server Error' },
+    },
+  },
+  POST: {
+    suggestedGasApiPostResponse: {
+      urlEndpoint: 'https://gas.api.cx.metamask.io/networks/1/suggestedGasFees',
+      response: { status: 200, message: 'Success' },
+      requestBody: { priorityFee: '2', maxFee: '2.000855333' },
+    },
+  },
+};
+```
+
+### Response Structure
+
+Mock responses are stored in individual JSON files for each API or service within the `mock-responses` folder, making them easier to maintain and manage. Each API service has its own JSON response file, such as `gasApiResponse.json` for gas-related responses and `ethpriceResponse.json` for Ethereum price responses. This organisation enables clear separation of mock data and simplifies updates or additions.
+
+**Example:** `gasApiResponse.json`
+```json
+{
+  "suggestedGasApiResponses": {
+    "error": {
+      "message": "Internal Server Error"
+    }
+  },
+  "suggestedGasFeesApiGanache": {
+    // ... detailed gas fee data ...
+  }
+}
+```
+
+## Logging
+
+The mock server logs response statuses and bodies to help track mocked requests, making debugging more straightforward.
+
+## Using Mock Testing Effectively
+
+
+### When to Use Mocks:
```

Comment:
It may be nice to have a few examples. One area I know we don't do a good job that this will help support is error handling on features when APIs go down.